noxfile.py

```diff
@@ -45,122 +43,62 @@
 maxpython = max(supported_python_versions)
 minpython = min(supported_python_versions)
 
-current_python = f"{sys.version_info.major}.{sys.version_info.minor}"
-
 # The documentation should be build always using the same version of
 # Python, which should be the latest version of Python supported by Read
 # the Docs. Because Read the Docs takes some time to support new
 # releases of Python, we should not link docpython to maxpython.
 
 docpython = "3.12"
 
+current_python = f"{sys.version_info.major}.{sys.version_info.minor}"
 nox.options.sessions: list[str] = [f"tests-{current_python}(skipslow)"]
-nox.options.default_venv_backend = "uv|virtualenv"
-
-running_on_ci = os.getenv("CI")
 
+nox.options.default_venv_backend = "uv|virtualenv"
 
-def _get_requirements_filepath(
-    category: Literal["docs", "tests", "all"],
-    version: Literal["3.11", "3.12", "3.13"],
-    resolution: Literal["highest", "lowest-direct", "lowest"] = "highest",
-    os_platform: Literal["linux", "macos", "windows"] | None = None,
-) -> str:
-    """
-    Return the file path to the requirements file.
+uv_sync = ("uv", "sync", "--no-progress", "--frozen")
 
-    Parameters
-    ----------
-    category : str
-        The name of the optional dependency set, as defined in
-        :file:`pyproject.toml`.
+running_on_ci = os.getenv("CI")
+running_on_rtd = os.environ.get("READTHEDOCS") == "True"
 
-    version : str
-        The supported version of Python.
+uv_requirement = "uv >= 0.6.1"
```

Comment:
`uv` is not listed as a package dependency, should it be?  If `uv` is included as a dependency, then this line can get out of sync easily.

Reply:
I have been wondering the same thing!  I've seen it recommended that `uv` be installed completely independently of Python, which makes sense since `uv` can be used to install and manage multiple versions of Python at once. 

`uv` needs to get installed first because it's what is used to install the dependencies in the tests. I put the `uv` requirement in here explicitly as a safeguard so that running `nox -s docs` will work even if `uv` hasn't been installed separately. 

I'm going to need to take a bit more time to think about it! 😅